src/utils/extract-customer-name.ts

```diff
@@ -0,0 +1,69 @@
+import { Cart, Order } from "@medusajs/medusa";
```

Comment:
nit(non blocking): we could simplify that with something along that line: wdyt? It is really a proposal and not something to necessary do, it also allow to update the props on which to look for pretty easily and to centralise the logic which is duplicate for each prop. just wanted to get your opinion :)
```ts
const extractCustomerName = (obj?: Cart | Order): string => {
  if (!obj) {
    return "N/A"
  }

  let firstName = ""
  let lastName = ""

  ;["customer", "shipping_address", "billing_address"].some((prop) => {
    firstName = obj[prop]?.first_name
    lastName =  obj[prop]?.last_name
    return !!(lastName || firstName)
  })

  if (firstName && lastName) {
    return `${firstName} ${lastName}`
  }

  if (firstName || lastName) {
    return firstName ?? lastName
  }

  if (obj.email) {
    return obj.email
  }

  return "N/A"
}
}

export default extractCustomerName
```